src/ipld/cid_hashset.rs

```diff
@@ -1,16 +1,17 @@
 // Copyright 2019-2023 ChainSafe Systems
 // SPDX-License-Identifier: Apache-2.0, MIT
 
-use crate::ipld::CidHashMap;
+use crate::utils::cid::SmallCid;
+use ahash::HashSet;
 use cid::Cid;
 
 #[derive(Default, Clone)]
-pub struct CidHashSet(CidHashMap<()>);
+pub struct CidHashSet(HashSet<SmallCid>);
```

Comment:
I am not sold on this. It wastes a lot of memory. Let's discuss this in slack.

Reply:
https://github.com/ChainSafe/forest/issues/3498

Reply:
I don't feel this is resolved.

Reply:
Blocked on #3502